src/sparse/impl/KokkosSparse_spmv_struct_impl.hpp

```diff
@@ -947,429 +930,334 @@ struct SPMV_MV_Struct_Transpose_Functor {
                                 static_cast<y_value_type> (val * m_x(iRow, k)));
           }
         }
-      }
-    }
+      });
+    });
   }
 };
 
-  template<class AMatrix,
-           class XVector,
-           class YVector,
-           int doalpha,
-           int dobeta,
-           bool conjugate>
-  struct SPMV_MV_Struct_LayoutLeft_Functor {
-    typedef typename AMatrix::execution_space            execution_space;
-    typedef typename AMatrix::non_const_ordinal_type     ordinal_type;
-    typedef typename AMatrix::non_const_value_type       A_value_type;
-    typedef typename YVector::non_const_value_type       y_value_type;
-    typedef typename Kokkos::TeamPolicy<execution_space> team_policy;
-    typedef typename team_policy::member_type            team_member;
-    typedef typename YVector::non_const_value_type       coefficient_type;
-
-    const coefficient_type alpha;
-    AMatrix m_A;
-    XVector m_x;
-    const coefficient_type beta;
-    YVector m_y;
-    //! The number of columns in the input and output MultiVectors.
-    ordinal_type n;
-    ordinal_type rows_per_thread;
-
-    SPMV_MV_Struct_LayoutLeft_Functor (const coefficient_type& alpha_,
-                                       const AMatrix& m_A_,
-                                       const XVector& m_x_,
-                                       const coefficient_type& beta_,
-                                       const YVector& m_y_,
-                                       const ordinal_type rows_per_thread_) :
-      alpha (alpha_),
-      m_A (m_A_), m_x (m_x_), beta (beta_), m_y (m_y_), n (m_x_.extent(1)),
-      rows_per_thread (rows_per_thread_)
-    {}
-
-    template<int UNROLL>
-    KOKKOS_INLINE_FUNCTION void
-    strip_mine (const team_member& dev, const ordinal_type& iRow, const ordinal_type& kk) const
-    {
-      y_value_type sum[UNROLL];
-
-#ifdef KOKKOS_ENABLE_PRAGMA_IVDEP
-#pragma ivdep
-#endif
-#ifdef KOKKOS_ENABLE_PRAGMA_UNROLL
-#pragma unroll
-#endif
-      for (int k = 0; k < UNROLL; ++k) {
-        sum[k] = Kokkos::Details::ArithTraits<y_value_type>::zero ();
-      }
+template<class AMatrix,
+         class XVector,
+         class YVector,
+         int doalpha,
+         int dobeta,
+         bool conjugate>
+struct SPMV_MV_Struct_LayoutLeft_Functor {
```

Comment:
So same comment as in the previous file, by the way we have a bunch of performance data on this and my experiments in the internal wiki I believe. I feel we should at least run the performance numbers at least once to see where this lands.